r/tests/testthat/test-dplyr-join.R

```diff
@@ -141,21 +141,25 @@ test_that("Error handling", {
 # TODO: casting: int and float columns?
 
 test_that("right_join", {
-  compare_dplyr_binding(
-    .input %>%
-      right_join(to_join, by = "some_grouping") %>%
-      collect(),
-    left
-  )
+  for (keep in c(TRUE, FALSE)) {
+    compare_dplyr_binding(
+      .input %>%
+        right_join(to_join, by = "some_grouping", keep = keep) %>%
+        collect(),
+      left
+    )
+  }
 })
 
 test_that("inner_join", {
-  compare_dplyr_binding(
-    .input %>%
-      inner_join(to_join, by = "some_grouping") %>%
-      collect(),
-    left
-  )
+  for (keep in c(TRUE, FALSE)) {
+    compare_dplyr_binding(
+      .input %>%
+        inner_join(to_join, by = "some_grouping") %>%
+        collect(),
+      left
+    )
+  }
```

Comment:
As there are only two things to iterate through here, mind including them separately instead of in a for loop? Just makes it easier to read in the logs if we have a failing test IMO.

Reply:
Yes, that was an issue. I fixed that just before I requested your review. It should be good now.